Задача 4

Huper12345 opened the pull request.

Huper12345 committed:
add branch for task 5 and push js files

Huper12345 committed:
edit js files

Huper12345 committed:
edit js files

Huper12345 committed:
edit readme.md

Huper12345 committed:
add function V paralelp

Huper12345 committed:
fix js problems

Huper12345 committed:
add function calc paralelp

Huper12345 committed:
add function calcVpyr

Huper12345 committed:
upgrade functions

Huper12345 committed:
add use strict

Huper12345 committed:
add additional func to calcVcube

Huper12345 committed:
add additional functions calcVpyr, calcVPparalelp

Huper12345 committed:
add additional func to calcVPparalelp

Huper12345 committed:
edit app js

Huper12345 committed:
add function buyPhone

Huper12345 committed:
add additional task to function buyphone

Huper12345 committed:
Fix mistakes in JS

Huper12345 committed:
Update app.js

Huper12345 committed:
Update readme.md

Huper12345 committed:
Update app.js

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+function proposeBuyphone() {
+    let message = confirm( 'Желаете купить телефон?' );
+
+if (message == true) {
```
не используем "нестрогое равенство" это очень плохая практика
всегда и везде применяем `===`

Huper12345 replied:
Поправил

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+proposeBuyphone();
+
+function proposeBuyphone() {
+    let message = confirm( 'Желаете купить телефон?' );
```
в ситуациях когда переменная не будет меняться стоит использовать `const`
это даст понять тому кто читает код, что переменная не меняется в дальнейшем

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+"use strict";
```
👍

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+        let message = confirm ( `Вы можете купить ${possibleBuy.toFixed(0)} телефонов. Желаете продолжить?` );
+
+        if (message == true) {
+            for ( let i = 0; i < balance ; balance = balance - phoneSum ) {
```
интересное применение цикла `for`
его точно не пустят в прод, но нешаблонность решения хороший знак 👍 
не пустят потому что `let i = 0 ` ни как не изменялось и вообще не использовалось в цикле
а значит и сама конструкция тут не нужна.
вместо неё напрашивается `while`

halavasty replied:
@Huper12345  залей сюда правку по циклю хочу увидеть какое условие будет у `while`

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+                        
+                        let buyAddaccesories = confirm ( 'Желаете купить дополнительные аксесуары на остаток средств на балансе?');
+                        
+                        if(buyAddaccesories == true) {
```
раз всё равно нужны будут правки, то и тут можно

`buyAddaccesories` само по себе будет `true` или `false` 
нет смысла дополнительно проверять является ли оно `true` 
```js
if(buyAddaccesories)
``
достаточно

из негативного отмечу `==` не используем нестрогое сравнение. только строгое!

halavasty commented on app.js:
```diff
@@ -0,0 +1,93 @@
+
+                            let acsessories = balance / phoneAccesories;
+                           
+                            for (let i = 0; i < balance; balance = balance - phoneAccesories) {
```
аналогично первому циклу

halavasty reviewed (commented):
Однозначно круто, что уже сейчас используешь разнообразные проверки на ошибки ввода 👍 
задачу дорабатываем

Huper12345 committed:
Исправил циклы на while, строгое равенство

Вот в чем странность while на последней итерации выбрасывает еще 1 alert с минусовым значением, я думал возможно это из за того что в условии "balance > 0" стоит 0 и когда проходит проверка а у нас допустим осталось 3$ то он еще раз выбрасывает alert, но если подставить в условие константу phoneAccesories (balance > phoneAccesories) она равняется 5 то в теории все должно было бы работать, но не работает), при этом если поставить просто 5 то работает. 
Решение такое, я скопировал константу в переменную и все заработало. Вопрос в следующем, почему цикл while выбрасывает мне еще 1 alert если я использую константу phoneAccesories?

Еще можно было бы просто убрать из for переменную i оставить вот так -  for (; balance > 0; balance = balance - phoneSum).
